op-proposer/proposer/service.go

```diff
@@ -203,6 +210,17 @@ func (ps *ProposerService) initL2ooAddress(cfg *CLIConfig) error {
 	return nil
 }
 
+func (ps *ProposerService) initDGF(cfg *CLIConfig) error {
+	l2ooAddress, err := opservice.ParseAddress(cfg.DGFAddress)
+	if err != nil {
+		return nil
```

Comment:
Why is `nil` returned here on an error? Same above in `initL2ooAddress` 🤔 

Is it to support both modi of operation? Maybe it's then cleaner to branch above in `initFromCLIConfig` depending on which address is set, and then only call either `initDGF` or `initL2ooAddress`?

Reply:
Good question; This was copied from the old behavior. Agree the branch above would be cleaner.